bgpd/bgp_route.c

```diff
@@ -1850,6 +1850,16 @@ int subgroup_announce_check(struct bgp_node *rn, struct bgp_path_info *pi,
 		if (!bgp_outbound_policy_exists(peer, filter))
 			return 0;
 
+	/* draft-ietf-idr-deprecate-as-set-confed-set
+	 * Filter routes having AS_SET or AS_CONFED_SET in the path.
+	 * Eventually, This document (if approved) updates RFC 4271
+	 * and RFC 5065 by eliminating AS_SET and AS_CONFED_SET types,
+	 * and obsoletes RFC 6472.
+	 */
+	if (peer->bgp->reject_as_sets == BGP_REJECT_AS_SETS_ENABLED)
+		if (aspath_check_as_sets(attr->aspath))
```

Comment:
I am worried about route aggregation scenarios. Are we breaking AS_SET for route aggregation too, thoughts ?

Reply:
yep, this is what the draft says:
```
Upon receipt of
   such messages, conformant BGP speakers SHOULD use the "Treat-as-
   withdraw" error handling behavior as per [RFC7606].
```
and:
```
 Network operators MUST NOT locally generate any new announcements
   containing AS_SET or AS_CONFED_SET.  If they have announced routes
   with AS_SET or AS_CONFED_SET in them, then they SHOULD withdraw those
   routes and re-announce routes for the aggregate or component prefixes
   (i.e., the more-specific routes subsumed by the previously aggregated
   route) without AS_SET or AS_CONFED_SET in the updates.
```